ambry-store/src/main/java/com.github.ambry.store/StoreMessageReadSet.java

```diff
@@ -171,7 +194,16 @@ public long writeTo(int index, WritableByteChannel channel, long relativeOffset,
     long startOffset = options.getOffset() + relativeOffset;
```

Comment:
Could these two lines (`sizeToRead` and the trace message) be moved inside the `transferTo()` case (as this is only applicable to that case?).